source/common/http/async_client_impl.h

```diff
@@ -164,7 +164,7 @@ class AsyncStreamImpl : public AsyncClient::Stream,
   AccessLog::RequestInfo& requestInfo() override { return request_info_; }
   const std::string& downstreamAddress() override { return EMPTY_STRING; }
   void continueDecoding() override { NOT_IMPLEMENTED; }
-  const Buffer::Instance* decodingBuffer() override { return decoding_buffer_; }
+  const Buffer::Instance* decodingBuffer() override { NOT_IMPLEMENTED; }
```

Comment:
this is going to just crash. Can you throw an actual EnvoyException (which will also end up crashing the process but indirectly) and add a test for this case please.